tests/helpers.py

```diff
@@ -1,89 +1,45 @@
+from sqlalchemy import func
+from sqlalchemy.engine import Engine
+from sqlalchemy.inspection import inspect
 
 
-def expected_data_within_get_response(
-    test_client,
-    endpoint: str,
-    expected_data,
-    exclude_paths=None,
-    exclude_regex_paths=None,
-):
-    """
-    Given a endpoint and expected content,
-    check to see if response contains expected data
-
-    Args:
-        test_client: A flask test client
-        endpoint (str): The GET request endpoint
-        expected_data: The content we expect to find
-        exclude_paths: DeepDiff dict paths to exclude from diff errors
-        exclude_regex_paths: DeepDiff dict regex paths
-            to exclude from diff errors
-
-    """
-    response = test_client.get(endpoint, follow_redirects=True)
-    response_data = json.loads(response.data)
-
-    diff = DeepDiff(
-        expected_data,
-        response_data,
-        exclude_paths=exclude_paths,
-        exclude_regex_paths=exclude_regex_paths,
-    )
+@contextmanager
+def no_gather_sql():
+    @event.listens_for(Engine, "before_cursor_execute", retval=True)
+    def mark_as_ignore(
+        conn, cursor, statement, parameters, context, executemany
+    ):
+        statement = statement + " --IGNORE"
+        return statement, parameters
 
-    error_message = f"Expected data does not match response: {str(diff)}"
-    assert diff == {}, error_message
+    yield
 
+    event.remove(Engine, "before_cursor_execute", mark_as_ignore)
 
-def put_response_return_200(test_client, endpoint):
-    """
-    Given a endpoint
-    check to see if returns a 200 success response
 
-    Args:
-        test_client: A flask test client
-        endpoint (str): The PUT request endpoint
+@no_gather_sql()
+def get_random_row(table):
+    """get_random_row Uses a database-side select to get a random
+    row. Does this by using a random offset with range (1, number of rows)
 
+    :param table: Sqlalchemy mapper object
+    :return: A random row from the given mapper.
     """
 
-    response = test_client.put(endpoint, follow_redirects=True)
-    assert response.status_code == 200
-
-
-def post_data(test_client, endpoint: str, data: dict):
-    """Given an endpoint and data, check to see if response contains expected data
-
-    Args:
-        test_client: A flask test client
-        endpoint (str): The POST request endpoint
-        data (dict): The content to post to the endpoint provided
-    """
-
-    response = test_client.post(
-        endpoint,
-        data=json.dumps(data),
-        content_type="application/json",
-        follow_redirects=True,
+    primary_key_name = inspect(table).primary_key[0].name
+    primary_key_column = getattr(table, primary_key_name)
+    return (
+        db.session.query(table)
+        .offset(
+            func.floor(
+                func.random()
+                * db.session.query(func.count(primary_key_column))
+            )
+        )
+        .limit(1)
+        .one()
```

Comment:
does limit(1) and one() do the same thing?

Reply:
limit refers to the sql query itself. It adds the LIMIT clause to the query. It gurentees that only a single result will be returned when pg executes the query.

One is a python side validator which will confirm that only one result was returned and extract it for us to use. 

Removing limit(1) causes an exception to be raised.